### PR TITLE
Keyboard/grandiceps

### PR DESCRIPTION
**Pull request by vattern**
## Description
Add new split ARM keyboard using F411 blackpills. Similar to sofle with underglow.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/tkw/grandiceps/rules.mk [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output
+
+WS2812_DRIVER = pwm
+OPT_DEFS += -DSTM32_DMA_REQUIRED=TRUE
+
+SPLIT_KEYBOARD = yes
+SERIAL_DRIVER = usart
+
+DEFAULT_FOLDER = tkw/grandiceps
```
Comment (zvecr) [inline keyboards/tkw/grandiceps/rules.mk:26]:
```suggestion
```

File: keyboards/tkw/grandiceps/rules.mk [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+OLED_DRIVER_ENABLE = yes
+
+# MCU name
+MCU = STM32F411
+
```
Comment (zvecr) [inline keyboards/tkw/grandiceps/rules.mk:31]:
```suggestion
```

File: keyboards/tkw/grandiceps/rules.mk [hunk 3]
```diff
@@ -0,0 +1,33 @@
+# Build Options
```
Comment (zvecr) [inline keyboards/tkw/grandiceps/rules.mk:1]:
```suggestion
```suggestion
# MCU name
MCU = STM32F411

# Build Options
```

File: keyboards/tkw/grandiceps/info.json [hunk 4]
```diff
@@ -0,0 +1,20 @@
+{
+    "keyboard_name": "grandiceps",
+    "url": "https://github.com/vattern/grandiceps",
+    "maintainer": "vattern",
+    "manufacturer": "tkw",
+    "width": 16.5,
+    "height": 5.25,
+    "layouts": {
+        "LAYOUT": {
+            "key_count": 60,
```
Comment (zvecr) [inline keyboards/tkw/grandiceps/info.json:10]:
```suggestion
```

File: keyboards/tkw/grandiceps/config.h [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#pragma once
+
+#include "config_common.h"
+
+#undef USE_I2C
+#undef SSD1306OLED
+
```
Comment (zvecr) [inline keyboards/tkw/grandiceps/config.h:22]:
```suggestion

```

File: keyboards/tkw/grandiceps/config.h [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+
+#define MATRIX_COL_PINS { B0, A7, A3, A5, A4, A2 }
+#define MATRIX_ROW_PINS { B12, A6, B13, B9, B8 }
+#define MATRIX_COL_PINS_RIGHT { B0, A7, A3, A5, A4, A2 }
+#define MATRIX_ROW_PINS_RIGHT  { B12, A6, B13, B9, B8 }
+/* COL2ROW, ROW2COL*/
+#define DIODE_DIRECTION COL2ROW
+#define MASTER_RIGHT
+// #define SPLIT_HAND_PIN B3
```
Comment (zvecr) [inline keyboards/tkw/grandiceps/config.h:43]:
```suggestion

```

File: keyboards/tkw/grandiceps/config.h [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+
+#define MATRIX_IO_DELAY 5
+#define TAP_CODE_DELAY 10
+
+#define ENCODERS_PAD_A { B14 }
+#define ENCODERS_PAD_B { B15 }
+
+#define RGB_DI_PIN B1
+#define RGBLED_NUM 16
+//#define RGBLED_SPLIT
```
Comment (zvecr) [inline keyboards/tkw/grandiceps/config.h:55]:
```suggestion
```

File: keyboards/tkw/grandiceps/config.h [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+#define DEBOUNCE 5
+
+
+/* disable these deprecated features by default */
+#define NO_ACTION_MACRO
+#define NO_ACTION_FUNCTION
+
+/* Bootmagic Lite key configuration */
+#define BOOTMAGIC_LITE_ROW 0
+#define BOOTMAGIC_LITE_COLUMN 0
```
Comment (zvecr) [inline keyboards/tkw/grandiceps/config.h:76]:
```suggestion
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/tkw/grandiceps/config.h [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+
+
+
+#define MATRIX_COL_PINS { B0, A7, A3, A5, A4, A2 }
+#define MATRIX_ROW_PINS { B12, A6, B13, B9, B8 }
+#define MATRIX_COL_PINS_RIGHT { B0, A7, A3, A5, A4, A2 }
+#define MATRIX_ROW_PINS_RIGHT  { B12, A6, B13, B9, B8 }
+/* COL2ROW, ROW2COL*/
+#define DIODE_DIRECTION COL2ROW
+#define MASTER_RIGHT
```
Comment (drashna) [inline keyboards/tkw/grandiceps/config.h:42]: I'm not a fan of having this defined by default.  No master set means it defaults to master left, and allows users to specify other options.

File: keyboards/tkw/grandiceps/rules.mk (hunk 3 above)
Comment (drashna) [reply keyboards/tkw/grandiceps/rules.mk:1]: (edited to remove duplicate suggestion markdown)

**Review by drashna — APPROVED** [review]

File: keyboards/tkw/grandiceps/readme.md [hunk 10]
```diff
@@ -0,0 +1,15 @@
+# grandiceps
+
+![grandiceps](https://i.imgur.com/FMtsboV.jpg)
```
Comment (fauxpark) [inline keyboards/tkw/grandiceps/readme.md:3]:
```suggestion
![grandiceps](https://i.imgur.com/FMtsboVl.jpg)
```

File: keyboards/tkw/grandiceps/readme.md [hunk 11]
```diff
@@ -0,0 +1,15 @@
...
+
+ARM split keyboard with RGB underglow and encoders.
+
+* Keyboard Maintainer: [vattern](https://github.com/vattern)
+* Hardware Supported: f411 blackpill
+* Hardware Availability: [grandiceps](https://github.com/vattern/grandiceps)
+
+Make example for this keyboard (after setting up your build environment):
+
+    qmk flash -kb tkw/grandiceps -km default
```
Comment (fauxpark) [inline keyboards/tkw/grandiceps/readme.md:13]:
```suggestion
    make tkw/grandiceps:default

Flashing example for this keyboard:

    make tkw/grandiceps:default:flash
```

File: keyboards/tkw/grandiceps/rules.mk [hunk 12]
```diff
@@ -0,0 +1,29 @@
+# MCU name
+MCU = STM32F411
+# Build Options
```
Comment (fauxpark) [inline keyboards/tkw/grandiceps/rules.mk:3]:
```suggestion

# Build Options
```

File: keyboards/tkw/grandiceps/rules.mk [hunk 13]
```diff
@@ -0,0 +1,29 @@
...
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+ENCODER_ENABLE = yes        # Enable rotary encoder support
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output
```
Comment (fauxpark) [inline keyboards/tkw/grandiceps/rules.mk:20]:
```suggestion
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = yes           # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
ENCODER_ENABLE = yes        # Enable rotary encoder support
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
KEYBOARD_SHARED_EP = yes    # Free up some extra endpoints - needed if console+mouse+extra
```

File: keyboards/tkw/grandiceps/keymaps/default/keymap.c [hunk 14]
```diff
@@ -0,0 +1,466 @@
...
+ *            |      |      |      |      |/       /         \      \ |      |      |      |      |
+ *            `----------------------------------'           '------''---------------------------'
+ */
+
+[_QWERTY] = LAYOUT( \
+  KC_ESC,   KC_1,   KC_2,    KC_3,    KC_4,    KC_5,                     KC_6,    KC_7,    KC_8,    KC_9,    KC_0,  KC_MINS, \
+  KC_DEL,   KC_Q,   KC_W,    KC_E,    KC_R,    KC_T,                     KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,  KC_BSPC, \
+  KC_TAB,   KC_A,   KC_S,    KC_D,    KC_F,    KC_G,                     KC_H,    KC_J,    KC_K,    KC_L, KC_SCLN,  KC_QUOT, \
+  KC_LSFT,  KC_Z,   KC_X,    KC_C,    KC_V,    KC_B, KC_MUTE,     XXXXXXX,KC_N,    KC_M, KC_COMM,  KC_DOT, KC_SLSH,  KC_RSFT, \
+                 KC_LGUI,KC_LALT,KC_LCTRL, KC_LOWER, KC_SPC,      KC_ENT,  KC_RAISE, KC_RCTRL, KC_RALT, KC_RGUI \
```
Comment (fauxpark) [inline keyboards/tkw/grandiceps/keymaps/default/keymap.c:65]:
```suggestion
[_QWERTY] = LAYOUT(
  KC_ESC,   KC_1,   KC_2,    KC_3,    KC_4,    KC_5,                     KC_6,    KC_7,    KC_8,    KC_9,    KC_0,  KC_MINS,
  KC_DEL,   KC_Q,   KC_W,    KC_E,    KC_R,    KC_T,                     KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,  KC_BSPC,
  KC_TAB,   KC_A,   KC_S,    KC_D,    KC_F,    KC_G,                     KC_H,    KC_J,    KC_K,    KC_L, KC_SCLN,  KC_QUOT,
  KC_LSFT,  KC_Z,   KC_X,    KC_C,    KC_V,    KC_B, KC_MUTE,     XXXXXXX,KC_N,    KC_M, KC_COMM,  KC_DOT, KC_SLSH,  KC_RSFT,
                 KC_LGUI,KC_LALT,KC_LCTRL, KC_LOWER, KC_SPC,      KC_ENT,  KC_RAISE, KC_RCTRL, KC_RALT, KC_RGUI
```

File: keyboards/tkw/grandiceps/keymaps/default/keymap.c [hunk 15]
```diff
@@ -0,0 +1,466 @@
...
+ *            |      |      |      |      |/       /         \      \ |      |      |      |      |
+ *            `----------------------------------'           '------''---------------------------'
+ */
+
+[_COLEMAK] = LAYOUT( \
+  KC_GRV,   KC_1,   KC_2,    KC_3,    KC_4,    KC_5,                      KC_6,    KC_7,    KC_8,    KC_9,    KC_0,  KC_GRV, \
+  KC_ESC,   KC_Q,   KC_W,    KC_F,    KC_P,    KC_G,                      KC_J,    KC_L,    KC_U,    KC_Y, KC_SCLN,  KC_BSPC, \
+  KC_TAB,   KC_A,   KC_R,    KC_S,    KC_T,    KC_D,                      KC_H,    KC_N,    KC_E,    KC_I,    KC_O,  KC_QUOT, \
+  KC_LSFT,  KC_Z,   KC_X,    KC_C,    KC_V,    KC_B, KC_MUTE,      XXXXXXX,KC_K,    KC_M, KC_COMM,  KC_DOT, KC_SLSH,  KC_RSFT, \
+                 KC_LGUI,KC_LALT,KC_LCTRL,KC_LOWER, KC_ENT,      KC_SPC,  KC_RAISE, KC_RCTRL, KC_RALT, KC_RGUI \
```
Comment (fauxpark) [inline keyboards/tkw/grandiceps/keymaps/default/keymap.c:88]:
```suggestion
[_COLEMAK] = LAYOUT(
  KC_GRV,   KC_1,   KC_2,    KC_3,    KC_4,    KC_5,                      KC_6,    KC_7,    KC_8,    KC_9,    KC_0,  KC_GRV,
  KC_ESC,   KC_Q,   KC_W,    KC_F,    KC_P,    KC_G,                      KC_J,    KC_L,    KC_U,    KC_Y, KC_SCLN,  KC_BSPC,
  KC_TAB,   KC_A,   KC_R,    KC_S,    KC_T,    KC_D,                      KC_H,    KC_N,    KC_E,    KC_I,    KC_O,  KC_QUOT,
  KC_LSFT,  KC_Z,   KC_X,    KC_C,    KC_V,    KC_B, KC_MUTE,      XXXXXXX,KC_K,    KC_M, KC_COMM,  KC_DOT, KC_SLSH,  KC_RSFT,
                 KC_LGUI,KC_LALT,KC_LCTRL,KC_LOWER, KC_ENT,      KC_SPC,  KC_RAISE, KC_RCTRL, KC_RALT, KC_RGUI
```

File: keyboards/tkw/grandiceps/keymaps/default/keymap.c [hunk 16]
```diff
@@ -0,0 +1,466 @@
...
+ *            | LGUI | LAlt | LCTR |LOWER | /Enter  /       \Space \  |RAISE | RCTR | RAlt | RGUI |
+ *            |      |      |      |      |/       /         \      \ |      |      |      |      |
+ *            `----------------------------------'           '------''---------------------------'
+ */
+[_LOWER] = LAYOUT( \
+  _______,   KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,                       KC_F6,   KC_F7,   KC_F8,   KC_F9,  KC_F10,  KC_F11,\
+  KC_GRV,    KC_1,    KC_2,    KC_3,    KC_4,    KC_5,                        KC_6,    KC_7,    KC_8,    KC_9,    KC_0,  KC_F12, \
+  _______, KC_EXLM,   KC_AT, KC_HASH,  KC_DLR, KC_PERC,                       KC_CIRC, KC_AMPR, KC_ASTR, KC_LPRN, KC_RPRN, KC_PIPE, \
+  _______,  KC_EQL, KC_MINS, KC_PLUS, KC_LCBR, KC_RCBR, _______,       _______, KC_LBRC, KC_RBRC, KC_SCLN, KC_COLN, KC_BSLS, _______, \
+                       _______, _______, _______, _______, _______,       _______, _______, _______, _______, _______\
```
Comment (fauxpark) [inline keyboards/tkw/grandiceps/keymaps/default/keymap.c:109]:
```suggestion
[_LOWER] = LAYOUT(
  _______,   KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,                       KC_F6,   KC_F7,   KC_F8,   KC_F9,  KC_F10,  KC_F11,
  KC_GRV,    KC_1,    KC_2,    KC_3,    KC_4,    KC_5,                        KC_6,    KC_7,    KC_8,    KC_9,    KC_0,  KC_F12,
  _______, KC_EXLM,   KC_AT, KC_HASH,  KC_DLR, KC_PERC,                       KC_CIRC, KC_AMPR, KC_ASTR, KC_LPRN, KC_RPRN, KC_PIPE,
  _______,  KC_EQL, KC_MINS, KC_PLUS, KC_LCBR, KC_RCBR, _______,       _______, KC_LBRC, KC_RBRC, KC_SCLN, KC_COLN, KC_BSLS, _______,
                       _______, _______, _______, _______, _______,       _______, _______, _______, _______, _______
```

File: keyboards/tkw/grandiceps/keymaps/default/keymap.c [hunk 17]
```diff
@@ -0,0 +1,466 @@
...
+ *            | LGUI | LAlt | LCTR |LOWER | /Enter  /       \Space \  |RAISE | RCTR | RAlt | RGUI |
+ *            |      |      |      |      |/       /         \      \ |      |      |      |      |
+ *            `----------------------------------'           '------''---------------------------'
+ */
+[_RAISE] = LAYOUT( \
+  _______, _______ , _______ , _______ , _______ , _______,                           _______,  _______  , _______,  _______ ,  _______ ,_______, \
+  _______, KC_INS,   KC_PSCR, KC_APP,   XXXXXXX,   XXXXXXX,                           KC_PGUP, KC_PRVWD,   KC_UP,   KC_NXTWD,   KC_DLINE, KC_BSPC, \
+  _______, KC_LALT,  KC_LCTL, KC_LSFT,  XXXXXXX,   KC_CAPS,                           KC_PGDN, KC_LEFT,    KC_DOWN, KC_RGHT,    KC_DEL,   KC_BSPC, \
+  _______, KC_UNDO,  KC_CUT,  KC_COPY,  KC_PASTE,  XXXXXXX,  _______,       _______,  XXXXXXX, KC_LSTRT,   XXXXXXX, KC_LEND,    XXXXXXX, _______, \
+                         _______, _______, _______, _______, _______,       _______, _______, _______, _______, _______ \
```
Comment (fauxpark) [inline keyboards/tkw/grandiceps/keymaps/default/keymap.c:130]:
```suggestion
[_RAISE] = LAYOUT(
  _______, _______ , _______ , _______ , _______ , _______,                           _______,  _______  , _______,  _______ ,  _______ ,_______,
  _______, KC_INS,   KC_PSCR, KC_APP,   XXXXXXX,   XXXXXXX,                           KC_PGUP, KC_PRVWD,   KC_UP,   KC_NXTWD,   KC_DLINE, KC_BSPC,
  _______, KC_LALT,  KC_LCTL, KC_LSFT,  XXXXXXX,   KC_CAPS,                           KC_PGDN, KC_LEFT,    KC_DOWN, KC_RGHT,    KC_DEL,   KC_BSPC,
  _______, KC_UNDO,  KC_CUT,  KC_COPY,  KC_PASTE,  XXXXXXX,  _______,       _______,  XXXXXXX, KC_LSTRT,   XXXXXXX, KC_LEND,    XXXXXXX, _______,
                         _______, _______, _______, _______, _______,       _______, _______, _______, _______, _______
```

File: keyboards/tkw/grandiceps/keymaps/default/keymap.c [hunk 18]
```diff
@@ -0,0 +1,466 @@
...
+ *            | LGUI | LAlt | LCTR |LOWER | /Enter  /       \Space \  |RAISE | RCTR | RAlt | RGUI |
+ *            |      |      |      |      |/       /         \      \ |      |      |      |      |
+ *            `----------------------------------'           '------''---------------------------'
+ */
+  [_ADJUST] = LAYOUT( \
+  XXXXXXX , XXXXXXX,  XXXXXXX,  XXXXXXX,   XXXXXXX, XXXXXXX,                    RGB_MOD,  RGB_VAI, RGB_SAI, RGB_HUI, XXXXXXX, RGB_TOG, \
+  RESET  ,  XXXXXXX,  KC_QWERTY,KC_COLEMAK,CG_TOGG, XXXXXXX,                    RGB_RMOD, RGB_VAD, RGB_SAD, RGB_HUD, XXXXXXX, XXXXXXX, \
+  XXXXXXX , XXXXXXX,  CG_TOGG,  XXXXXXX,   XXXXXXX, XXXXXXX,                    XXXXXXX,  KC_VOLD, KC_MUTE, KC_VOLU, XXXXXXX, XXXXXXX, \
+  XXXXXXX , XXXXXXX,  XXXXXXX,  XXXXXXX,   XXXXXXX, XXXXXXX, XXXXXXX,  XXXXXXX, XXXXXXX,  KC_MPRV, KC_MPLY, KC_MNXT, XXXXXXX, XXXXXXX, \
+            _______, _______, _______,          _______, _______,         _______, _______,        _______, _______, _______ \
```
Comment (fauxpark) [inline keyboards/tkw/grandiceps/keymaps/default/keymap.c:151]:
```suggestion
  [_ADJUST] = LAYOUT(
  XXXXXXX , XXXXXXX,  XXXXXXX,  XXXXXXX,   XXXXXXX, XXXXXXX,                    RGB_MOD,  RGB_VAI, RGB_SAI, RGB_HUI, XXXXXXX, RGB_TOG,
  RESET  ,  XXXXXXX,  KC_QWERTY,KC_COLEMAK,CG_TOGG, XXXXXXX,                    RGB_RMOD, RGB_VAD, RGB_SAD, RGB_HUD, XXXXXXX, XXXXXXX,
  XXXXXXX , XXXXXXX,  CG_TOGG,  XXXXXXX,   XXXXXXX, XXXXXXX,                    XXXXXXX,  KC_VOLD, KC_MUTE, KC_VOLU, XXXXXXX, XXXXXXX,
  XXXXXXX , XXXXXXX,  XXXXXXX,  XXXXXXX,   XXXXXXX, XXXXXXX, XXXXXXX,  XXXXXXX, XXXXXXX,  KC_MPRV, KC_MPLY, KC_MNXT, XXXXXXX, XXXXXXX,
            _______, _______, _______,          _______, _______,         _______, _______,        _______, _______, _______
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

File: keyboards/tkw/grandiceps/chconf.h [hunk 19]
```diff
@@ -0,0 +1,714 @@
+/*
```
Comment (fauxpark) [inline keyboards/tkw/grandiceps/chconf.h:1]: Please run `qmk chibios-confmigrate` on these *conf.h files.

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

File: keyboards/tkw/grandiceps/chconf.h [hunk 20]
```diff
@@ -0,0 +1,27 @@
...
+ */
+
+/*
+ * This file was auto-generated by:
+ *    `qmk chibios-confupdate -i keyboards/tkw/grandiceps/chconf.h -r platforms/chibios/BLACKPILL_STM32_F411/configs/chconf.h`
+ */
+
+#pragma once
+
+#define CH_CFG_ST_FREQUENCY 100000
```
Comment (zvecr) [inline keyboards/tkw/grandiceps/chconf.h:24]: This is the default, so the whole file should be able to be deleted.

File: keyboards/tkw/grandiceps/grandiceps.h [hunk 21]
```diff
@@ -0,0 +1,40 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#pragma once
+
+#define _x_ KC_NO
+
```
Comment (zvecr) [inline keyboards/tkw/grandiceps/grandiceps.h:19]:
```suggestion

```

**Review by zvecr — CHANGES_REQUESTED** [review]

**Review by zvecr — APPROVED** [review]

**Pull request merged**